test/api.sh

```diff
@@ -256,29 +264,29 @@ step_6_test_batch_evaluation()
 step_7_test_delete()
 {
     # delete rules and distributions
-    shakedown DELETE "/api/v1/flags/$flag_key/rules/$rule_id_1" -H 'Content-Type:application/json'
+    _shakedown DELETE "/api/v1/flags/$flag_key/rules/$rule_id_1" -H 'Content-Type:application/json'
         status 200
 
-    shakedown DELETE "/api/v1/flags/$flag_key/rules/$rule_id_2" -H 'Content-Type:application/json'
+    _shakedown DELETE "/api/v1/flags/$flag_key/rules/$rule_id_2" -H 'Content-Type:application/json'
         status 200
 
     # delete flag and variants
-    shakedown DELETE "/api/v1/flags/$flag_key" -H 'Content-Type:application/json'
+    _shakedown DELETE "/api/v1/flags/$flag_key" -H 'Content-Type:application/json'
         status 200
 
     # delete segment and constraints
-    shakedown DELETE "/api/v1/segments/$segment_key" -H 'Content-Type:application/json'
+    _shakedown DELETE "/api/v1/segments/$segment_key" -H 'Content-Type:application/json'
         status 200
 }
 
 step_8_test_meta()
 {
-    shakedown GET "/meta/info"
+    _shakedown GET "/meta/info"
```

Comment:
similar to above, we may want to think about if we should protect all endpoints behind auth if we arent already, dont want to leak any data that users dont want leaked without auth

Reply:
Yeah, agreed. I think this endpoint should be authenticated.

Health and ready, less so. And there are always going to be some endpoints which must live outside.
e.g. authorize URL and callback for oauth. Those are either safe or authorized in their own way (callback authorizes using the provided `code` to ask the authenticating service).

Reply:
I realize now authenticating this particular part is going to require more work.
It is not hard (some code structure questions need answering) but so far, we have been authenticating at the gRPC level.

These endpoints are all HTTP only.

Reply:
Yeah maybe we could merge this and add a follow up to make `meta` authenticated?